wapitiCore/net/crawler.py

```diff
@@ -926,6 +926,52 @@ def iter_forms(self, autofill=True):
                 )
                 yield new_form
 
+    def find_login_form(self):
+        """Returns the login Request extracted from the Page, the username and password fields."""
+
+        for form in self.soup.find_all("form"):
+            username_field_idx = []
+            password_field_idx = []
+
+            for i, input_field in enumerate(form.find_all("input")):
+                input_type = input_field.attrs.get("type", "text").lower()
+                input_name = input_field.attrs.get("name", "undefined")
+                input_id = input_field.attrs.get("id", "undefined")
+                if input_type == "email":
+                    username_field_idx.append(i)
+
+                elif input_type != "password" and (
+                        any(field_name in input_name for field_name in ["mail", "user", "login", "name"]) or
+                        any(field_id in input_id for field_id in ["mail", "user", "login", "name"])
+                ):
+                    username_field_idx.append(i)
+
+                elif input_type == "password":
+                    password_field_idx.append(i)
+
+            # ensure login form
+            if len(username_field_idx) == 1 and len(password_field_idx) == 1:
+                inputs = form.find_all("input", attrs={"name": True})
+
+                url = self.make_absolute(form.attrs.get("action", "").strip() or self._url)
+                method = "POST"
```

Comment:
A form may use GET. In fact GET is the default method if not specified in the form `method` attribute.

It may seems stupid to deal with those cases as `-s` can be leveraged for such cases (like `-s http://target/login.php?user=test&pass=test`) but it may be useful to deal with it it the form have some CSRF tokens